src/ImageSharp/Memory/Allocators/ArrayPoolMemoryAllocator.Buffer{T}.cs

```diff
@@ -53,8 +53,13 @@ public override Span<T> GetSpan()
                 {
                     ThrowObjectDisposedException();
                 }
-
+#if SUPPORTS_CREATESPAN
+                ref byte r0 = ref MemoryMarshal.GetReference<byte>(this.Data);
```

Comment:
Small note - this is implicitly creating a `Span<byte>` from the array. When we add .NET 5 support we can optimize this even further by using `MemoryMarshal.GetArrayDataReference` instead and completely bypass all checks 😊

Reply:
Yeah... I was looking at that. I still don't know what to do re .NET 5 though. It's be our first target framework which isn't LTS.

Reply:
Ah, right. The way I see it, we could maybe do:
- Add a .NET 5 target to start preparing the codebase for .NET 6. When .NET 6 lands, immediately switch to it and drop .NET 5.
- Just hold our breath until .NET 6 lands and stick to .NET Core 3.1 as max until then.

I'd say it might be worth doing an initial investigation for starters, to see how many places in the lib could actually benefit from .NET 5 exclusive APIs that are not on .NET Core 3.1? Then we could decide based on that 🙂

Reply:
ARM intrinsics is a big one.

Reply:
Ah, right, right. Yeah that alone might be worth it. Properly supporting ARM64 gives a ton of visibility right now, plus being ready for .NET 6 would be very nice. And if we are to work on ARM64 right now anyway, I don't see the harm in publishing .NET 5 packages already, assuming we'll get a working build soon. After all, companies would likely remain on .NET Core 3.1 until .NET 6 is out, so those wouldn't be affected. And devs on .NET 5 now would be fine with having to jump to .NET 6 immediately anyway, after all they jsut did the same with .NET 5 right now. I'd say if you have time you could look into this and then when you have a branch that shows a nice boost on ARM64, consider shipping if it's really good? 😄